forge-ai/src/main/java/forge/ai/simulation/GameCopier.java

```diff
@@ -91,6 +91,15 @@ public Game makeCopy(PhaseType advanceToPhase, Player aiPlayer) {
             newPlayer.setLifeLostLastTurn(origPlayer.getLifeLostLastTurn());
             newPlayer.setLifeLostThisTurn(origPlayer.getLifeLostThisTurn());
             newPlayer.setLifeGainedThisTurn(origPlayer.getLifeGainedThisTurn());
+            newPlayer.setBlessing(origPlayer.hasBlessing());
+            newPlayer.setRevolt(origPlayer.hasRevolt());
+            newPlayer.setLibrarySearched(origPlayer.getLibrarySearched());
+            newPlayer.setSpellsCastLastTurn(origPlayer.getSpellsCastLastTurn());
+            for (int j = 0; j < origPlayer.getSpellsCastThisTurn(); j++) {
+                newPlayer.addSpellCastThisTurn();
+            }
+            newPlayer.setMaxHandSize(origPlayer.getMaxHandSize());
```

Comment:
was there a point where these hand sizes didn't work in the copy?
Because currently they are done only via static abilities, so I would have expected them to just be naturally carried over

Reply:
I know there are cards that say "you don't have a maximum hand size for the rest of the game", but I didn't personally test whether this is needed. I just added what looked to be missing.